lib/install/management/store_files.go

```diff
@@ -53,15 +53,37 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *metadata.Virtu
 	var emptyImages bool
 	var emptyVolumes bool
 	log.Infof("Removing images")
-	if emptyImages, err = d.deleteImages(ds, p); err != nil {
-		errs = append(errs, err.Error())
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
```

Comment:
@fdawg4l has some utility functions to create a URL from ds & path, and visa versa. I'd recommend using those - extending them if necessary.
The specific reason for this is the case of special characters in the datastore name (something I'd overlooked when suggesting using the Host field for the datastore) - at some point we may well change how dspaths are packed so using a utility function to extract datastore from URL and path from URL means only a single location to change if/when we change packing.